spec/components/result-view/plotly-spec.js

```diff
@@ -0,0 +1,123 @@
+"use babel";
+
+import { mount } from "enzyme";
+import _ from "lodash";
+import React from "react";
+
+import PlotlyTransform from "../../../lib/components/result-view/plotly";
+
+import plotly from "@nteract/plotly";
+
+function deepFreeze(obj) {
+  // Retrieve the property names defined on obj
+  const propNames = Object.getOwnPropertyNames(obj);
+
+  // Freeze properties before freezing self
+  propNames.forEach(name => {
+    const prop = obj[name];
+
+    // Freeze prop if it is an object
+    if (typeof prop === "object" && prop !== null) {
+      deepFreeze(prop);
+    }
+  });
+
+  // Freeze self (no-op if already frozen)
+  return Object.freeze(obj);
+}
+
+const figure = deepFreeze({
+  data: [
+    { x: [1999, 2000, 2001, 2002], y: [10, 15, 13, 17], type: "scatter" },
+    { x: [1999, 2000, 2001, 2002], y: [16, 5, 11, 9], type: "scatter" }
+  ],
+  layout: {
+    title: "Super Stuff",
+    xaxis: { title: "Year", showgrid: false, zeroline: false },
+    yaxis: { title: "Percent", showline: false },
+    height: "100px"
+  }
+});
+
+describe("PlotlyTransform", () => {
+  it("plots some data from an object", () => {
+    spyOn(plotly, "newPlot");
+    const plotComponent = mount(<PlotlyTransform data={figure} />);
+
+    const instance = plotComponent.instance();
+
+    expect(instance.shouldComponentUpdate({ data: "" })).toBeTruthy();
+    expect(plotly.newPlot).toHaveBeenCalledWith(
+      instance.plotDiv,
+      [
+        { x: [1999, 2000, 2001, 2002], y: [10, 15, 13, 17], type: "scatter" },
+        { x: [1999, 2000, 2001, 2002], y: [16, 5, 11, 9], type: "scatter" }
+      ],
+      {
+        title: "Super Stuff",
+        xaxis: { title: "Year", showgrid: false, zeroline: false },
+        yaxis: { title: "Percent", showline: false },
+        height: "100px"
+      },
+      {
+        modeBarButtonsToRemove: ["toImage"],
+        modeBarButtonsToAdd: [
+          {
+            name: "Download plot as a png",
+            icon: plotly.Icons.camera,
+            click: instance.downloadImage
+          }
+        ]
+      }
+    );
+  });
+
```

Comment:
I'm not so sure how useful some of the tests ported over from the monorepo would be. Do you think they are useful for us here?

Reply:
I think that they are useful since we will basically have to make sure we stay up to date with plotly and don't break this component